pennylane/_qubit_device.py

```diff
@@ -312,10 +312,22 @@ def execute(self, circuit, **kwargs):
         # apply all circuit operations
         self.apply(circuit.operations, rotations=circuit.diagonalizing_gates, **kwargs)
 
-        # generate computational basis samples
+        # generate computational basis samples and check for commutativity in sampled observables
         if self.shots is not None or circuit.is_sampled:
             self._samples = self.generate_samples()
 
+            sampled_obs = []
+            for o in circuit.measurements:
+                if o.return_type in (qml.measurements.Sample, qml.measurements.Counts):
+                    sampled_obs.append(o)
+
+            raw_sampled_ops = any(o.obs is None for o in sampled_obs)
```

Comment:
Could we get this information from the previous `for` loop via `if o.return_type in (qml.measurements.Sample, qml.measurements.Counts) and o.obs is None:`? Might also be good to stop at the first occurrence.

Reply:
Great catch!